Navbar Update

televandalist opened the pull request:
A couple things were bugging me and one thing led to another... 🤷‍♂️ 

Changes to the `Achievements` dropdown:
  - Adds `New Sets & Revisions` which leads [here](https://retroachievements.org/claimlist.php?s=9&f=8109).
  - Adds `Sets in Progress` and removes `Claim List` from the `Community` dropdown.
  - Adds `Most Requested Sets` and removes it from the `Manage` dropdown.

Changes to the `Community` dropdown:
  - Adds `RANews Home` and `RANews Issues List`
  - Adds `RAGuides Wiki`
  - Adds `Community Events` which links to [the hub](https://retroachievements.org/game/3105).
  - Adds `Developer Events` which links to [the hub](https://retroachievements.org/game/3273).
  - Adds `Hub List` which is [this page](https://retroachievements.org/gameList.php?s=6&c=100&f=1)
  - Adds `Central Hub` and links to [the hub](https://retroachievements.org/game/6914).
  - Removes `- Community` since it led to the same page as `Forums`
  - Removes `- Developers` since the Developer forum is rarely used for anything and shows up in the forum index anyway.
  - Renames `Forums` to `Forum Index`
  - Renames `Recent Posts` to `Recent Forum Posts`
  - Renames the tacky `+- Competitions` link to `Event Forums`
  - Adds a second column to the dropdown since it was getting crowded.

Adds a `Contact` dropdown with links to send a message to the following accounts:
  - [RAdmin](https://retroachievements.org/createmessage.php?t=RAdmin)
  - [DevCompliance](https://retroachievements.org/createmessage.php?t=DevCompliance)
  - [QATeam](https://retroachievements.org/createmessage.php?t=QATeam)
  - [RANews](https://retroachievements.org/createmessage.php?t=RANews)
  - [RAEvents](https://retroachievements.org/createmessage.php?t=RAEvents)
  - [DevQuest](https://retroachievements.org/createmessage.php?t=DevQuest)
  - [TheUnwanted](https://retroachievements.org/createmessage.php?t=TheUnwanted)
  - [QualityQuest](https://retroachievements.org/createmessage.php?t=QualityQuest)

Jamiras commented:
I don't like the two-column view for Community.
![image](https://user-images.githubusercontent.com/32680403/218268083-23214ace-4ffe-4ed2-9156-a28d395ff29f.png)

I also don't like the hubs being in there. I have never once wanted to navigate to a hub directly. "Community Events", "Developer Events", "Hub List" and "Central Hub" are all hub links. The first two I expected to take me to active events, but they show the entire history. If anything, they should be in the Games menu:
![image](https://user-images.githubusercontent.com/32680403/218268485-49d38044-f5d5-44d6-86ca-57b61a8854ad.png)

Then the only changes to the Community menu are the additional links to the RANews and RAGuides:
![image](https://user-images.githubusercontent.com/32680403/218268609-190a11b0-81af-4f2f-8803-318e8c2a901a.png)

Jamiras commented:
I have reservations against the Contact menu. It's not something I think should be featured so prominently in the header. Most sites that provide that sort of functionality have a single "Contact" link in the footer which launches a page where the user can decide who they want to contact. This additional page cold also describe each account so the user is more likely to contact the appropriate one.

Speaking of the footer, we also have a Community section there with links to the Global Ranking, RANews, Forum Index, and Users List. That could be removed in favor of the existing menu. Additionally, the Achievements footer section is redundant with three of the top-level menu items. And the Documentation footer section provides completely different links than the documentation links in the Community menu.

luchaos commented:
> I have reservations against the Contact menu

I agree; i was about to review it in a very similar way. Same for the other concerns.

Changing the main menu should be introduced after very good consideration of what the standard UX for users would be; streamlining the experience of interacting with the site and/or contacting staff. There might be issues I'm not aware of - I can see how most of your suggested changes might improve hypothetical issues. Do we need better guidance in particular for users to have them find ways to communicate with us @televandalist that came up in the past?

drisc commented:
> I have never once wanted to navigate to a hub directly.

That may be true for you but I've wanted an easy way to navigate to the major hubs on the site for a long while.

I understand wanting to just have them be a away to catagorize things in the sidebar but they are also a powerful tool for discovering new games, especially the themed hubs. Having a way to access specific hubs or even just the central one without having to find a game that is linked to it would be a usablity improvment in my eyes.

RALordAndrew commented:
Personally, I think the Series, Genre, and Central hubs would be good to include in the Games menu. It helps players find games they're interested in, and encourages developers to link appropriate hubs.

televandalist commented:
> I don't like the two-column view for Community.

I originally had more stuff in the both columns, which made it seem a bit more necessary. After condensing it a bit, it does seem a little pointless.

> I also don't like the hubs being in there. I have never once wanted to navigate to a hub directly. "Community Events", "Developer Events", "Hub List" and "Central Hub" are all hub links. The first two I expected to take me to active events, but they show the entire history. If anything, they should be in the Games menu:

We don't have any pages specifically for events or hubs, just the hub and event entries themselves. Both of them having their own home bases on the site would be preferable so this is really just a case of working with what we have at the moment. I can get behind hubs being in the Games menu, though.

 > I have reservations against the Contact menu. It's not something I think should be featured so prominently in the header. Most sites that provide that sort of functionality have a single "Contact" link in the footer which launches a page where the user can decide who they want to contact. This additional page cold also describe each account so the user is more likely to contact the appropriate one.

> Speaking of the footer, we also have a Community section there with links to the Global Ranking, RANews, Forum Index, and Users List. That could be removed in favor of the existing menu. Additionally, the Achievements footer section is redundant with three of the top-level menu items. And the Documentation footer section provides completely different links than the documentation links in the Community menu.

The contact menu is just another case of working with what we have. I'd prefer to have a support section on the site that contains all the necessary info. As for the footer, I forgot it about it since I usually don't have a reason to scroll all the way down. I'd be fine with scrapping most of it. 

> There might be issues I'm not aware of - I can see how most of your suggested changes might fix hypothetical issues. Do we need better guidance, in particular, for users so they can find ways to communicate with us @televandalist? Anything that came up in the past?

We have multiple sub-teams that serve important purposes, but they're virtually nonexistent outside of Discord and occasional forum posts. Some of these teams go dormant for a bit or barely get going at all, despite the need for them. Newer users tend to be unaware of them so issues end up not being reported, in a game/achievement wall comment that no one saw, or DM'd to someone who doesn't immediately address it because they get distracted easily (~me~). 

A couple very recent examples:

- A user messaged me asking me if I would fix a few typos in a set (which wasn't mine). They were unaware that we have a writing team specifically for that. 

- Hashing for [this game](https://retroachievements.org/game/11250) was broken for a couple months, causing it to be unplayable for several users. The only talk about it was on the wall and forum. I started looking into it after Ryudo mentioned something in our DuckStation testing channel. Found out that RAHasher wasn't hashing it properly, and asked @Sanaki to double-check. Something like 40 other games were affected by it so we reported it to Jamiras, who [fixed the issue shortly after](https://github.com/RetroAchievements/rcheevos/pull/216).

- This is more of an ongoing thing... but there are constant forum posts by users asking to link certain versions of a game to a set. We have a compatibility testing team, but I think only 2-3 users have actively participated. The team currently does not have someone maintaining or coordinating the work, despite it being an often-requested team and absolutely necessary for what we do. If a support/contact page exists, users would actually know the team exists. 

Not only are some users unaware of who to contact, we're also likely missing out on potential help.

So yeah, I think a support/contact page on site is necessary.

Soulsderfuchs commented:
I'd also like to say having a direct link to the central hub is extremely useful, as with the recent search update the hubs are very hard to find themselves on the search results this will help when finding them. Hopefully the search is also updated to take into account hub first instead of games.

televandalist commented:
- Changed `Community` back to one-column view.
- Added `Miscellaneous` header under `Games`
- Moved the hub links under `Games`
- Added `Genre & Subgenre` and `Series` central hub links
- Added `Nintendo DSi` and `Atari Jaguar CD` links, but with comment slashes
- Removed `RANews Issues List` link
- Removed `Contact` menu due to https://github.com/RetroAchievements/RAWeb/pull/1362

Jamiras commented on app_legacy/Helpers/render/layout.php:
```diff
@@ -105,16 +107,28 @@ function RenderToolbar(): void
+    echo "<li><a href='/game/9553'>Genre & Subgenre Hub</a></li>";
+    echo "<li><a href='/game/5771'>Series Hub</a></li>";
+    echo "<li><a href='/game/3105'>Community Events</a></li>";
+    echo "<li><a href='/game/3273'>Developer Events</a></li>";
```
Should `Community Events` and `Developer Events` also specify Hub? i.e. `Community Events Hub`?

televandalist replied:
Specified Hub for the events, but I'm wondering if any of them even need it since they're already under the Hub category 🤔

Jamiras commented on app_legacy/Helpers/render/layout.php:
```diff
@@ -105,16 +107,28 @@ function RenderToolbar(): void
     echo "<ul>";
+    echo "<li><a href='/claimlist.php?s=9&f=8109'>New Sets & Revisions</a></li>";
+    echo "<li><a href='/claimlist.php'>Sets in Progress</a></li>";
+    echo "<li><a href='/setRequestList.php'>Most Requested Sets</a></li>";
```
This is redundant with the `Most Requested` link in the Games menu. Should `New Sets` and `Sets in Progress` be moved there?

televandalist replied:
Got rid of `Most Requested` in the Achievements menu. I guess I overlooked that one. Moved the others to the Games menu since it makes sense.

Jamiras commented on app_legacy/Helpers/render/layout.php:
```diff
@@ -127,18 +141,18 @@ function RenderToolbar(): void
-    echo "<li><a href='/forum.php?c=7'>- Developers</a></li>";
-    echo "<li><a href='/forumposthistory.php'>Recent Posts</a></li>";
+    echo "<li><a href='https://news.retroachievements.org/'>RANews</li>";
+    echo "<li><a href='https://github.com/RetroAchievements/guides/wiki/'>RAGuides Wiki</li>";
```
I really think the forums should be the first items in the list and these should be moved down towards the documentation. I'd probably change the order of some of the other items too:
```
Forum Index
Event Forums
Recent Forum Posts
-----
User List
Global Ranking
Recent Masteries
Developer Stats
-----
RANews
RAGuides Wiki
-----
User Documentation
Developer Documentation
```

televandalist replied:
Resorted per the list

Jamiras commented on app_legacy/Helpers/render/layout.php:
```diff
@@ -105,16 +107,27 @@ function RenderToolbar(): void
     echo "<li><a href='#'>Achievements</a>";
     echo "<div>";
     echo "<ul>";
+    echo "<li class='divider'></li>";
```
there's nothing above this divider.
